pkg/cpu/cpu_linux.go

```diff
@@ -37,117 +37,109 @@ func (i *Info) load() error {
 	return nil
 }
 
-func ProcByID(procs []*Processor, id int) *Processor {
-	for pid := range procs {
-		if procs[pid].ID == id {
-			return procs[pid]
-		}
-	}
-	return nil
-}
-
-func CoreByID(cores []*ProcessorCore, id int) *ProcessorCore {
-	for cid := range cores {
-		if cores[cid].Index == id {
-			return cores[cid]
-		}
-	}
-	return nil
-}
-
 func processorsGet(ctx *context.Context) []*Processor {
-	procs := make([]*Processor, 0)
 	paths := linuxpath.New(ctx)
 
-	r, err := os.Open(paths.ProcCpuinfo)
-	if err != nil {
-		return nil
-	}
-	defer util.SafeClose(r)
-
-	// An array of maps of attributes describing the logical processor
-	procAttrs := make([]map[string]string, 0)
-	curProcAttrs := make(map[string]string)
-
-	// Parse /proc/cpuinfo
-	scanner := bufio.NewScanner(r)
-	for scanner.Scan() {
-		line := strings.TrimSpace(scanner.Text())
-		if line == "" {
-			// Output of /proc/cpuinfo has a blank newline to separate logical
-			// processors, so here we collect up all the attributes we've
-			// collected for this logical processor block
-			procAttrs = append(procAttrs, curProcAttrs)
-			// Reset the current set of processor attributes...
-			curProcAttrs = make(map[string]string)
-			continue
-		}
-		parts := strings.Split(line, ":")
-		key := strings.TrimSpace(parts[0])
-		value := strings.TrimSpace(parts[1])
-		curProcAttrs[key] = value
-	}
+	lps := logicalProcessorsFromProcCPUInfo(ctx)
+	// keyed by processor ID (physical_package_id)
+	procs := map[int]*Processor{}
 
-	// Iterate on /sys/devices/system/cpu/cpuN, not on /proc/cpuinfo
-	Entries, err := ioutil.ReadDir(paths.SysDevicesSystemCPU)
+	// /sys/devices/system/cpu pseudodir contains N number of pseudodirs with
+	// information about the logical processors on the host. These logical
+	// processor pseudodirs are of the pattern /sys/devices/system/cpu/cpu{N}
+	fnames, err := ioutil.ReadDir(paths.SysDevicesSystemCPU)
 	if err != nil {
-		return nil
+		ctx.Warn("failed to read /sys/devices/system/cpu: %s", err)
```

Comment:
ok, the rest of the function should be skipped as `fnames` is expected to be empty slice

Reply:
ah, good catch. I will re-add the `return nil`